relayer/pkg/chainlink/ocr2/providers.go

```diff
@@ -28,8 +28,8 @@ type configProvider struct {
 	lggr logger.Logger
 }
 
-func NewConfigProvider(chainID string, contractAddress string, url string, cfg starknet.Config, lggr logger.Logger) (*configProvider, error) {
-	chainReader, err := NewClient(chainID, url, lggr, cfg)
+func NewConfigProvider(chainID string, contractAddress string, basereader starknet.Reader, cfg starknet.Config, lggr logger.Logger) (*configProvider, error) {
```

Comment:
The `ConfigProvider` should depend just on `ocr2.Reader` as it just reads, doesn't write anything.

The `MedianProvider` on the other hand should depend on `ocr2.Client` which it uses for both reading and writing to the chain.

While these functions construct their own readers/clients (I'm lost here lol) now, ideally we do IoC and they depend on them, we provide/inject them.

Reply:
why does the `MedianProvider` need the `writer` component? afaik the only part that writes to the chain is the txm

Reply:
The `MedianProvider` provides the `ContractTransmitter` that depends on TXM.

So the `MedianProvider` either requests TXM and uses it to construct the `ContractTransmitter`, or it requests the client and uses it to construct the TXM, and in turn the `ContractTransmitter`.

As currently the TXM is provided externally by the Chain type, we'll probably inject the TXM here. In that case, it doesn't need a `writter` but just an additional `reader`.

Does that make sense?

Reply:
yup makes sense! I also realized that I forgot to connect the txm and and `ContractTransmitter` so I will add that implementation and then that should make more sense.

Reply:
ok! now `Chain.TxManager` is now used to pass the txm to `MedianProvider` => `ContractTransmitter`

the txmanager is started together with the `Chain` in core